### PR TITLE
updated Mongoid to 6.0.0

### PR DESCRIPTION
**Pull request by moodlemags**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ source 'http://rubygems.org'
 
 gemspec
 
-case version = ENV['MONGOID_VERSION'] || '5.0'
+case version = ENV['MONGOID_VERSION'] || '6.0'
 when /^6/
   gem 'mongoid', github: 'mongodb/mongoid'
```
Comment (dblock) [inline Gemfile]: This should now be a version locked at 6.0 and you might want to add a HEAD option via .travis.yml similar to how https://github.com/mongoid/mongoid-history does it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 * [#4](https://github.com/mongoid/mongoid-compatibility/pull/4): Use the `mongoid-danger` gem - [@dblock](https://github.com/dblock).
 * [#3](https://github.com/mongoid/mongoid-compatibility/pull/3): Added Danger, PR linter - [@dblock](https://github.com/dblock).
+* Updated to Mongoid 6.0.0 - [@moodlemags](https://github.com/moodlemags).
```
Comment (dblock) [inline CHANGELOG.md]: Make this "Compatibility with Mongoid 6.0", it also will need the #5 in there just like the other lines.

**Comment by dblock** [comment]
You have to look at build failures too. It's possible that we no longer can support older versions of Ruby or we need not to run a certain number of builds on older versions or Ruby (preferred). Either way iterate until it's green.

**Comment by dblock** [comment]
Still need the build to pass, probably by locking ActiveSupport to an older version for these specific versions of mongoid.

**Comment by moodlemags** [comment]
@dblock 👍  :)

File: Gemfile [hunk 3]
```diff
@@ -2,7 +2,9 @@ source 'http://rubygems.org'
 
 gemspec
 
-case version = ENV['MONGOID_VERSION'] || '5.0'
+case version = ENV['MONGOID_VERSION'] || '6.0'
+when 'HEAD'
+  gem 'mongoid', github: 'mongodb/mongoid'
 when /^6/
   gem 'mongoid', github: 'mongodb/mongoid'
```
Comment (dblock) [inline Gemfile]: This is still incorrect, should be `~> 6.0`, otherwise we're running tests against the next version.

**Comment by dblock** [comment]
Almost! See above.

**Comment by moodlemags** [comment]
@dblock Seems I was suffering from a serious case of overthinking the problem. It's now the same format as Mongoid history. Is there a way to see if everything checks without pushing?

**Comment by dblock** [comment]
You have to let Travis do its thing, but otherwise yes you have to do something like this for each:

````
export MONGOID_VERSION=4
rm Gemfile.lock
bundle
rake
```

**Comment by dblock** [comment]
`rubocop -a` ;)

**Comment by moodlemags** [comment]
oooh wow that makes it much simpler to check

**Comment by dblock** [comment]
Indeed, now just fix the build and we should be good to merge this.

**Comment by dblock** [comment]
Also read up on "squashing your commits", you should be able to do this here, and end up with 1 nice commit. [This](http://code.dblock.org/2015/08/31/getting-out-of-your-first-git-mess.html) might also be helpful.

**Comment by dblock** [comment]
Code looks great. Squashing didn't work :) Try following my article?

**Comment by dblock** [comment]
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html could work too

**Comment by moodlemags** [comment]
You weren't supposed to look yet! Had a call. Squashing away now 🐿

**Comment by moodlemags** [comment]
Just kidding. It makes sense but it isn't cooperating, I have them squashed on my side but tis not reflecting up here

**Comment by dblock** [comment]
You have to push the changes up. Is it failing? You might need a force push with `git push origin maggie -f`.

**Comment by moodlemags** [comment]
It was the -f! I didn't think it would work, it kept saying no new files to update. I guess technically nothing was updated within the files. That probably took longer than it should have..? But it was super fun, I am ready for more please.

**Comment by dblock** [comment]
Merged. Thanks!

**Comment by dblock** [comment]
Another quick one: https://github.com/mongoid/mongoid-grid_fs was just moved to the mongoid org. It has some broken links (like the build badge). Its build is failing and it could use Danger (see https://github.com/mongoid/danger).

**Pull request merged**